yt_dlp/extractor/vrt.py

```diff
@@ -231,100 +293,76 @@ class VrtNUIE(VRTBaseIE):
+            }, note='Logging in', errnote='Login failed')
         self._authenticated = True
+        return
 
     def _real_extract(self, url):
         display_id = self._match_id(url)
         parsed_url = urllib.parse.urlparse(url)
-        details = self._download_json(
-            f'{parsed_url.scheme}://{parsed_url.netloc}{parsed_url.path.rstrip("/")}.model.json',
-            display_id, 'Downloading asset JSON', 'Unable to download asset JSON')['details']
-
-        watch_info = traverse_obj(details, (
-            'actions', lambda _, v: v['type'] == 'watch-episode', {dict}), get_all=False) or {}
-        video_id = join_nonempty(
-            'episodePublicationId', 'episodeVideoId', delim='$', from_dict=watch_info)
-        if '$' not in video_id:
-            raise ExtractorError('Unable to extract video ID')
-
-        vrtnutoken = self._download_json(
-            'https://token.vrt.be/refreshtoken', video_id, note='Retrieving vrtnutoken',
-            errnote='Token refresh failed')['vrtnutoken'] if self._authenticated else None
-
-        video_info = self._call_api(video_id, 'vrtnu-web@PROD', vrtnutoken)
-
-        if 'title' not in video_info:
-            code = video_info.get('code')
-            if code in ('AUTHENTICATION_REQUIRED', 'CONTENT_IS_AGE_RESTRICTED'):
-                self.raise_login_required(code, method='password')
-            elif code in ('INVALID_LOCATION', 'CONTENT_AVAILABLE_ONLY_IN_BE'):
-                self.raise_geo_restricted(countries=['BE'])
-            elif code == 'CONTENT_AVAILABLE_ONLY_FOR_BE_RESIDENTS_AND_EXPATS':
-                if not self._authenticated:
-                    self.raise_login_required(code, method='password')
-                self.raise_geo_restricted(countries=['BE'])
-            raise ExtractorError(code, expected=True)
-
-        formats, subtitles = self._extract_formats_and_subtitles(video_info, video_id)
+
+        # 1. Obtain/refresh 'vrtnu-site_profile' tokens
+        self._request_webpage('https://www.vrt.be/vrtnu/sso/login', None, note='Getting tokens', errnote='Failed to get tokens')
+
+        # 2. Perform GraphQL query to obtain video metadata
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': f'Bearer {self._get_cookies("https://www.vrt.be").get("vrtnu-site_profile_at").value}'
+        }
+
+        data = {
+            'operationName': 'VideoPage',
+            'query': self._VIDEOPAGE_QUERY,
+            'variables': {
+                'pageId': f'{parsed_url.path.rstrip("/")}.model.json'
+            }
+        }
+
+        metadata = self._download_json(
+            'https://www.vrt.be/vrtnu-api/graphql/v1',
+            display_id, 'Downloading asset JSON', 'Unable to download asset JSON', headers=headers, data=json.dumps(data).encode())['data']['page']
+
+        video_id = metadata['episode']['watchAction']['streamId']
+        ld_json = json.loads(metadata['ldjson'][1])
```

Comment:
Things like this should be non-fatal to still allow downloading videos even if say the metadata format changes.

Reply:
OK, but failure to extract video_id should remain fatal, right?

Reply:
Yeah, the video_id can stay fatal, but the ld_json doesn't need to be.